Add a style guide and chapter template

LuisHeinzlmeier opened the pull request:
- Updated `CONTRIBUTING.md` to give contributors a style guide. 
- Added a chapter template to help contributors quickly and effectively create a chapter for our book.
- Closes #337

LuisHeinzlmeier commented:
The second link should work when this PR is merged (text in `CONTRIBUTING.md`). 
`We’ve also prepared a chapter [template](/jupyter-book/template/template.ipynb) of which you can view the html output [here](https://www.sc-best-practices.org/template/template.html) (Note the boxes in the text!).`
At the moment, you can check out the template html in the preview [here](https://www.sc-best-practices.org/pr-preview/pr-367/template/template.html).

seohyonkim commented:
Do we perhaps want to add something about using
```{admonition} title
:class: dropdown
```
for the dropdowns?

LuisHeinzlmeier commented:
> Do we perhaps want to add something about using admonition with
> 
> ```
> :class: dropdown
> ```
> 
> for the dropdowns?

You mean because they are then visible in the PDF?

Zethson reviewed (changes requested):
Thank you very much @LuisHeinzlmeier !

This is already going into the right direction but I think we have to agree a bit on the style. From what I can see, your template is currently more of an example than a guide like template.

There's a few improvements that I think are necessary:

1. I found the template to be very hard to understand without some introduction in the beginning. You can write some markdown at the beginning that explain what this template is, how it can be used, what features it shows, ... . People are supposed to remove that cell of course.
2. I don't think that we should have grid and grid cards. They don't render right?
3. We should refer to the contributing guide in the template and ask users to read that first.
4. We should link to the jupyter book documentation regularly where relevant.
5. A lot of the example text and code is unnecessarily long. Keep it to 1-2 lines of text and code. Anything more is noise.
6. Mention figures and how to create them.
7. Mention lamindb and what our expectations here are. Please ask me for help if this is unclear.

Generally, if I were to read through the template, I should be able to copy it and work on my own chapter. For that we need the links to the external material like the guide + jupyter book and show off a few more jupyter book features where relevant. What you have needs to be more beginner oriented.

Let's have one more round of iteration based on my (and Seo's) comments and then we can go more into details.

LuisHeinzlmeier commented:
Thank you very much for the feedback @Zethson!

1. That's a good idea. I have now created an introductory cell. ✅ Tell me if I am on the right track:
```
Welcome to the single cell best practices template! This template is designed to help you quickly and effectively create well-structured, clear, and consistent chapters for our book. It serves as a guide, presenting our rules, general tips, and ready-to-use code and markdown snippets that can be easily adapted to your needs.
Make sure to start by reviewing our Contributing Guide. It contains essential guidelines for writing and formatting ours chapters. Besides that, only the most important features such as text formatting, interactive elements, images and code cells are shown in this template. The primary source for detailed documentation is the [Jupyter Book](https://jupyterbook.org/en/stable/intro.html) documentation, but [MySTmd](https://mystmd.org) or [Sphinx](https://www.sphinx-doc.org/en/master/) can also help.
```
2. No, grid and grid cards only render when we build the html. Will contributors read the template in their development environment (e.g., VS), GitHub, or a browser? As rendered html or pure markdown?
3. Added this to the introductory cell! ✅ 
4. -6. I will add/update that!
7. I think I need help with the question of what expectations we **exactly** have of lamindb.

I will let you know when I have adjusted your feedback!

Zethson commented:
Users will read them in their favorite IDE/editor like vscode

Zethson commented:
> Welcome to the single cell best practices template! This template is designed to help you quickly 
> and effectively create well-structured, clear, and consistent chapters for our book. It serves as 
> a guide, presenting our rules, general tips, and ready-to-use code and markdown snippets that 
> can be easily adapted to your needs.
> Make sure to start by reviewing our Contributing Guide. It contains essential guidelines for writing and 
> formatting ours chapters. Besides that, only the most important features such as text formatting, 
> interactive elements, images and code cells are shown in this template. The primary source for 
> detailed documentation is the [Jupyter Book](https://jupyterbook.org/en/stable/intro.html) 
> documentation, but [MySTmd](https://mystmd.org) or 
> [Sphinx](https://www.sphinx-doc.org/en/master/) can also help.

You are absolutely on the right track. It's a tiny bit verbose but that can easily be improved. Thank you!

LuisHeinzlmeier commented:
@Zethson, I implemented all the feedback!

I think I just need help with the question of what expectations we **exactly** have of lamindb.

Zethson commented on CONTRIBUTING.md:
```diff
@@ -1,30 +1,49 @@
-We highly welcome community contributions and encourage contributions.
+## Our philosophy
+
+We aim for this book to be a major resource for introducing people to the field of single-cell sequencing data analysis and ensuring that newcomers and experienced analysis do things right.
```
```suggestion
We aim for this book to be a resource for introducing people to single-cell and spatial data analysis while ensuring that newcomers and experienced analysts alike do things properly.
```

Zethson commented on CONTRIBUTING.md:
```diff
@@ -1,30 +1,49 @@
+However, as mentioned [above](#our-philosophy), these recommendations are held to high standards.
+
+If you're proposing new tutorials, corrections, or extensions, we strongly recommend opening an issue first to discuss your ideas with us.
+We're always eager to learn about the latest developments and will likely have follow-up questions — but we're genuinely excited to collaborate.
```
```suggestion
We're always eager to learn about the latest developments and are happy to discuss the current state of best practices.
```

LuisHeinzlmeier replied:
Adopted!

Zethson commented on CONTRIBUTING.md:
```diff
@@ -1,30 +1,49 @@
+If you're proposing new tutorials, corrections, or extensions, we strongly recommend opening an issue first to discuss your ideas with us.
+We're always eager to learn about the latest developments and will likely have follow-up questions — but we're genuinely excited to collaborate.
+
+We also encourage contributions in the form of [quizzes and flashcards](#create-custom-quizzes-and-flashcards).
```
I get it but I'm not sure whether we want to add like 20 more questions per chapter. It'd be a lot to maintain and potentially verbose.

LuisHeinzlmeier replied:
I understand we should discuss this. I just took the content from the old `CONTRIBUTING.md` and updated it. What is the approximate/maximum number of questions we want per chapter? 5-15?

seohyonkim replied:
I think less than 5 would look good, but for bigger chapters 5-8 sounds good!

Zethson commented on CONTRIBUTING.md:
```diff
@@ -1,30 +1,49 @@
+We're always eager to learn about the latest developments and will likely have follow-up questions — but we're genuinely excited to collaborate.
+
+We also encourage contributions in the form of [quizzes and flashcards](#create-custom-quizzes-and-flashcards).
+Our collection of self-assessment questions is always growing, and community support in expanding them is greatly appreciated.
```
```suggestion
```

LuisHeinzlmeier replied:
Adopted!

Zethson commented on CONTRIBUTING.md:
```diff
@@ -1,30 +1,49 @@
+
+We also encourage contributions in the form of [quizzes and flashcards](#create-custom-quizzes-and-flashcards).
+Our collection of self-assessment questions is always growing, and community support in expanding them is greatly appreciated.
+The same applies to these contributions: Just open an issue and let's get in touch!
```
```suggestion
The same applies to these contributions: Just open an issue and let's get in touch!
```

I'd rephrase this to a general request for issues to discuss proposals.

Zethson commented on CONTRIBUTING.md:
```diff
@@ -1,30 +1,49 @@
-The following depicts the rough structure of the book inside the `jupyter_book` folder.
-Beyond several configuration files, all chapters are stored inside corresponding folders, such as for example, `conditions`.
-Inside, the corresponding notebooks, together with the associated `reference.bib` and `environment.yml` files, are stored.
+The structure of the book is organized within the `jupyter_book` folder.
```
Weird sentence. I'd suggest that we rephrase this to the source of the book being in that folder or so

Zethson commented on CONTRIBUTING.md:
```diff
@@ -73,9 +93,170 @@ make clean
+### Rules
+
+- Place each sentence on its own line to make reviewing easier.
+- Use dropdowns whenever possible so that the reader is not overwhelmed by the content.
```
```suggestion
- Use dropdowns so that the reader is not overwhelmed by the content.
```

I'll weaken this a bit although I pushed for this.

LuisHeinzlmeier replied:
Adopted!

Zethson reviewed (commented):
Concerning Lamin, I'm rambling some random points that need to be put into text:

1. 1-3 sentences of what lamindb is and what we're using it for
2. When making contributions that got merged, ask to be added to the `theislab/sc-best-practices` instance to rerun the notebook and have the latest version also in the instance.
3. Ensure that all datasets are loaded from the instance. Add some information about the scripts folder where the ingest scripts for the datasets are stored.

Zethson commented:
> Got it!
> 
> Do we also want to use `lamin` for sth like this: ![Bildschirmfoto 2025-05-26 um 07 40 45](https://private-user-images.githubusercontent.com/182783669/447428412-054b15ea-3124-4237-ae23-5007bcee8bd6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDgyNDQwNTgsIm5iZiI6MTc0ODI0Mzc1OCwicGF0aCI6Ii8xODI3ODM2NjkvNDQ3NDI4NDEyLTA1NGIxNWVhLTMxMjQtNDIzNy1hZTIzLTUwMDdiY2VlOGJkNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUyNlQwNzE1NThaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kYjQ2NzEyYTU4Y2E4MGFhM2RiYjQ0NDgwY2IyN2E0MjYwNjVmYjhhNTlhOGM2ZWNjOTgwNWFiYjk3ZjU5NGRlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.IN0Eu6hapCJgyW9fPTUhBwCUyReZrHojG4Q6frp5V0Y)

Yes, in this case for consistency. Also because then we don't depend on mudatadatasets

LuisHeinzlmeier commented:
```
## Data access

We are currently still working on making all datasets accessible.
The problem here isn't data protection or other terrible reasons, 
but rather the requirement to host them somewhere and ensuring 
that the correct versions are used per notebook.
```

I have noticed that we can omit this text now, can't we? This is solved with `lamindb`.

LuisHeinzlmeier commented:
But apart from that. If you, @Zethson,  are happy with the content I created for `lamindb`, we can merge it! 🚀 🥳

LuisHeinzlmeier commented:
Perfect, no problem! Thank you, Lukas!

Zethson reviewed (approved):
This is really great work. Thank you!
